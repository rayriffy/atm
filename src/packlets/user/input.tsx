import { Heading } from '@chakra-ui/react'

import { useWithdrawInput } from './useWithdrawInput'

export const Input = () => {
  const { value } = useWithdrawInput()
  return (
    <Heading fontSize="7xl" pt={12} textAlign="right">
      Â£{value.toLocaleString()}
    </Heading>
  )
}
