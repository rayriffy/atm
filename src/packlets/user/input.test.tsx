import { render } from '@testing-library/react'
import { vi, describe, it, expect, beforeEach } from 'vitest'

import { Input } from './input'

const { useWithdrawInputSpy } = vi.hoisted(() => ({
  useWithdrawInputSpy: vi.fn(() => ({
    value: 0,
  })),
}))

vi.mock('./useWithdrawInput', () => ({
  useWithdrawInput: useWithdrawInputSpy,
}))

describe('<Input />', () => {
  beforeEach(() => {
    vi.clearAllMocks()
  })

  it('should render input', () => {
    const { container } = render(<Input />)

    expect(container).toMatchSnapshot()
  })

  it('should render comma in value', () => {
    useWithdrawInputSpy.mockReturnValue({ value: 123456789 })

    const { getByText } = render(<Input />)

    expect(getByText('Â£123,456,789')).toBeDefined()
  })
})
