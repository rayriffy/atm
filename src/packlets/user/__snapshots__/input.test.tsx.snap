// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Input /> > should render input 1`] = `
<div>
  <h2
    class="chakra-heading css-120tsxn"
  >
    Â£
    0
  </h2>
</div>
`;
